packages/perseus/src/util/graphie.ts

```diff
@@ -1411,6 +1434,57 @@ const labelDirections = {
     "above left": [-1.0, -1.0],
 } as const;
 
+// A range for a Graphie can be specified in one of three forms:
+// - A pair of intervals, [[xMin, xMax], [yMin, yMax]]. This is the
+//   "normalized" form.
+// - A pair of magnitudes for the x and y ranges, respectively. E.g. [5, 10]
+//   is equivalent to [[-5, 5], [-10, 10]].
+// - A single magnitude to be used for both x and y. E.g. 10 is equivalent to
+//   [[-10, 10], [-10, 10]]
+type RangeSpecifier = [Interval, Interval] | Coord | number;
+
+// Overload normalizeRange to specify that it only returns undefined if its
+// argument is undefined.
+export function normalizeRange(range: RangeSpecifier): [Interval, Interval];
+export function normalizeRange(range: RangeSpecifier | undefined): [Interval, Interval] | undefined;
```

Comment:
**question**: wouldn't this overload simply be this?

```suggestion
export function normalizeRange(range: undefined): undefined;
```

Reply:
I thought about adding that overload, but it doesn't seem useful. If you know the input is always `undefined`, why call `normalizeRange`? 😂 

`RangeSpecifier | undefined` allows you to pass in a value that *might* be undefined, on the condition that you can handle getting an undefined value back. The other overload (without undefined) guarantees that if you don't pass undefined, you won't get undefined back.

Reply:
Ah, that makes sense. I hadn't thought about the types that _might_ be `undefined`, but might not.